addons/mass_mailing/models/mass_mailing_blacklist.py

```diff
@@ -0,0 +1,147 @@
+
+    # method that will be overriden in the model that inherit the mixin
+    # to know on which field the _compute_is_blacklisted() must depend
+    def _blacklist_get_email_field_name(self):
+        return []
+
+    # Creates the depends expression to work with two fields, one known and one from model that inherit the mixin.
+    def _blacklist_get_depends(self):
+        if self._blacklist_get_email_field_name():
+            [email_field] = self._blacklist_get_email_field_name()
+            return ['blacklist_id.active', email_field]
+        else:
+            return ['blacklist_id.active']
+
+    # blacklist_id creates an explicit link to the blacklist model
+    # in order to know which email in model that inherits the mixin is impacted by changes in blacklist model.
+    # an non stored fields cannot be used as dependence
+    # except if there is a search method on it (that inverse the many2one dependence)
+    # so each time the blacklist is modified 'created, deleted or set to active/inactive,
+    # the blacklist_id of the impacted records is re_computed and the is_blacklist field too
+    blacklist_id = fields.Many2one('mail.mass_mailing.blacklist', store=False, search="_blacklist_search_blacklist_id",
+                                   string='Blacklist id')
+
+    # Note :
+    # With store = False, the access to the field is not evaluated because the fields is not in DB.
+    # With store = True, the access to the field can be evaluated.
+    is_blacklisted = fields.Boolean(string='Blacklist', compute="_blacklist_compute_is_blacklisted", store=True,
+        help="If the email address is on the blacklist, the contact won't receive mass mailing anymore, from any list",
+        groups="base.group_user")
+    blacklist_reason = fields.Char(compute="_blacklist_compute_is_blacklisted", store=True)
+
+    @api.model
+    def _blacklist_search_blacklist_id(self, operator, value):
+        if self._blacklist_get_email_field_name():
+            [email_field] = self._blacklist_get_email_field_name()
+            # implement search with domain [('blacklist_id', 'in', value)]
+            assert operator == 'in' and isinstance(value, list)
+            black_recs = self.env['mail.mass_mailing.blacklist'].browse(value)
+            emails = black_recs.mapped('email')
+            return [(email_field, 'in', emails)]
+        # else:
+        #     return []
+
+    @api.depends(lambda self: self._blacklist_get_depends())
+    def _blacklist_compute_is_blacklisted(self):
+        [email_field] = self._blacklist_get_email_field_name()
+        blacklist = self.env['mail.mass_mailing.blacklist'].sudo().search([('email', 'in', [item[email_field] for item in self])]).read(['email', 'reason', 'active'])
+        for rec in self:  # Optimisation: create indexed dict sorted by email and lookup in dict to get email and reason
+            email = rec[email_field]
+            item = next((x for x in blacklist if email == x['email']), None)
```

Comment:
maybe an alternative solution: 
((x for x in blacklist if email == x['email']) or [None])[0]
or, if we can have more than one result
((x for x in blacklist if email == x['email'])[:1] or [None])[0]

Just a personal opinion to make it more readable, just because next seems weird to me

Reply:
Or like the comment says, transform it to a dict
something like that:
```
items = dict[(values['email'],values) for values in blacklist ]
...
item = items.get("email", None)
```